File: 241227/신기한 타일 뒤집기/strange-flipping-tiles.java
import java.io.*;
import java.util.*;

public class Main {
    public static final int OFFSET = 100000, WHITE = 1, BLACK = 2;
    public static int[] tiles = new int[OFFSET * 2 + 1];
    public static int white = 0, black = 0;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int inputCount = Integer.parseInt(br.readLine());

        int cur = OFFSET;
        for (int i = 0; i < inputCount; i++) {
            String[] input = br.readLine().split(" ");
            int flipCount = Integer.parseInt(input[0]);
            String direction = input[1];

            if (direction.equals("L")) {
                while (flipCount > 0) {
                    tiles[cur] = WHITE;
                    flipCount--;
                    if (flipCount > 0) cur--;
                }
            } else {
                while (flipCount > 0) {
                    tiles[cur] = BLACK;
                    flipCount--;
                    if (flipCount > 0) cur++;
                }
            }
        }

        white = (int) Arrays.stream(tiles).filter(ele -> ele == WHITE).count();
        black = (int) Arrays.stream(tiles).filter(ele -> ele == BLACK).count();
        System.out.println(white + " " + black);
    }
}

File: 241227/신기한 타일 뒤집기/strange-flipping-tiles.py
n = int(input());
comm = [];
for i in range(4):
    m, d = input().split(); 
    comm.append((m,d));

now = 0;
l = 0;
r = 0;
for m,d in comm:
    m = int(m);
    if(d == "R"):
        now += m;
        if(now>=r): r=now;
    if(d == "L"):
        now -= m;
        if(now<=l): l=now;
print(r-now, now-l);
